Emby.Server.Implementations/Data/SqliteItemRepository.cs

```diff
@@ -6164,76 +6178,74 @@ public List<MediaAttachment> GetMediaAttachments(MediaAttachmentQuery query)
 
             if (query.Index.HasValue)
             {
-                cmdText += "AND AttachmentIndex=@AttachmentIndex";
+                cmdText += " AND AttachmentIndex=@AttachmentIndex";
             }
 
             cmdText += " order by AttachmentIndex ASC";
 
-            using (var connection = GetConnection(true))
+            var list = new List<MediaAttachment>();
+            using var connection = GetConnection(true);
+            using (var statement = PrepareStatement(connection, cmdText))
             {
-                var list = new List<MediaAttachment>();
+                statement.TryBind("@ItemId", query.ItemId.ToByteArray());
 
-                using (var statement = PrepareStatement(connection, cmdText))
+                if (query.Index.HasValue)
                 {
-                    statement.TryBind("@ItemId", query.ItemId.ToGuidBlob());
-
-                    if (query.Index.HasValue)
-                    {
-                        statement.TryBind("@AttachmentIndex", query.Index.Value);
-                    }
-
-                    foreach (var row in statement.ExecuteQuery()) {
-                        list.Add(GetMediaAttachment(row));
-                    }
+                    statement.TryBind("@AttachmentIndex", query.Index.Value);
                 }
 
-                return list;
+                foreach (var row in statement.ExecuteQuery())
+                {
+                    list.Add(GetMediaAttachment(row));
+                }
             }
+
+            return list;
         }
 
         public void SaveMediaAttachments(Guid id, List<MediaAttachment> attachments, CancellationToken cancellationToken)
         {
             CheckDisposed();
             if (id == Guid.Empty)
             {
-                throw new ArgumentNullException(nameof(id));
+                throw new ArgumentException(nameof(id));
             }
 
             if (attachments == null)
             {
                 throw new ArgumentNullException(nameof(attachments));
             }
 
-            using (var connection = GetConnection())
+            cancellationToken.ThrowIfCancellationRequested();
+
+            using var connection = GetConnection();
+            connection.RunInTransaction(db =>
             {
-                connection.RunInTransaction(db =>
-                {
-                    var itemIdBlob = id.ToGuidBlob();
+                var itemIdBlob = id.ToByteArray();
 
-                    db.Execute("delete from mediaattachments where ItemId=@ItemId", itemIdBlob);
+                db.Execute("delete from mediaattachments where ItemId=@ItemId", itemIdBlob);
 
-                    InsertMediaAttachments(itemIdBlob, attachments, db);
+                InsertMediaAttachments(itemIdBlob, attachments, db);
```

Comment:
again, why `InsertMediaAttachements()` doesn't get a `cancellationToken`? cannot it be cancelled?

Reply:
I see no answer to my question here, why do you resolve it without answering?

Reply:
I know see it was fixed... I would be grateful if you could say "fixed" next time :)

Reply:
No problem, I'll leave a note in the future.